Tutorial.playground/Contents.swift

```diff
@@ -0,0 +1,433 @@
+//:
+//: * `Pipe`s can both `yield` and `await` values and they model stream
+//: transformations
+//:
+//: * `Effect`s can neither `yield` nor `await` and they model non-streaming
+//: components
+//:
+//: You can connect these components together in four separate ways which
+//: parallel the four above types:
+//:
+//: * `for_` handles `yield`s
+//:
+//: * `>~` handles `await`s
+//:
+//: * `>->` handles both `yield`s and `await`s
+//:
+//: * `>>-` handles return values
+//:
+//: As you connect components their types will change to reflect inputs and
+//: outputs that you`ve fused away.  You know that you`re done connecting things
+//: when you get an `Effect`, meaning that you have handled all inputs and
+//: outputs.  You run this final `Effect` to begin streaming.
+
+import Swiftz
+import Aquifer
+
+// MARK: - Producers
+
+//: `Producer`s are effectful streams of input.  Specifically, a `Producer` is a
+//: Type that extends any other type with a new `yield` command. This `yield` 
+//: command lets you send output downstream to an anonymous handler, decoupling
+//: how you generate values from how you consume them.
+
+//: As an aside: Swift does not *technically* allow for the definition of polymorphic
+//: typealiases like `Producer`.  Instead, `Aquifer` uses a number of polymorphic enums
+//: with typealiases inside (the `.T` in all of the types presented hereafter).  We 
+//: specifically chose to use enums with no cases so there would be no option to instantiate
+//: them.  This way, they are markers and nothing more.
+ 
+//: The following `stdinByLine` `Producer` shows how to incrementally read in
+//: `String`s from standard input and `yield` them downstream, terminating
+//: gracefully when reaching the end of the input:
+
+import class Foundation.NSFileHandle
+
+//                                   +----------  A `Producer` that yields `String`s
+//                                   |
+//                                   |        +-- Every action has a return value.
+//                                   |        |   This action returns `()` when finished
+//                                   v        v
+public func stdinByLine() -> Producer<String, ()>.T {
+	let handle = NSFileHandle.fileHandleWithStandardInput() // Grab the handle
+	if handle.isAtEndOfFile { // If we`re at the end of the line, stop.
+		return pure(())
+	} else {
+		return yield(handle.readLine) >>- { _ in fromHandle(handle) } // Otherwise `yield` the line and loop.
+	}
+}
+
+//: `yield` emits a value, suspending the current `Producer` until the value is
+//: consumed.  If nobody consumes the value (which is possible) then `yield`
+//: never returns.  You can think of `yield` as having the following type:
+//:
+//:     func yield<A>(value : A) -> Producer<A, ()>.T
+//:
+//: The true type of `yield` is actually more general and powerful.  Throughout
+//: the tutorial I will present type signatures like this that are simplified at
```

Comment:
The pronouns thing is irking me more and more.  I prefer to think of TypeLift as a 'we'.